### ISSUE TITLE
Bizarre behavior with igraph plotting in jupyter lab

### ISSUE THREAD

**Issue by jvail** [jupyterlab/jupyterlab#10121]
I found a very bizarre behavior with plotting in jupyter lab (3.0.14), igraph (0.9.1) and google chrome (90.0.4430.72) both on linux and windows: Whenever I plot a graph it tries to render the complete jupyter lab UI, it seems. **Does not happen in firefox or classical notebooks**.

I can only guess that it might be related to the most recent chrome update because nothing else had changed in my environment.
See also the discussion here https://github.com/igraph/python-igraph/issues/388#issue-862418155

![image](https://user-images.githubusercontent.com/6503378/115349010-60862c80-a1c4-11eb-9915-d6f2853d3cde.png)

**Comment by ntamas** [comment]
To add some context to the issue: the result of `igraph.plot()` is a `Plot` object, whose `_repr_svg_()` method returns an SVG representation in a tuple, paired with `{"isolated": True}`. The latter is needed because igraph uses Cairo as the plotting backend, and Cairo re-uses XML IDs in subsequent plots, so non-isolated plots would get mixed-up glyphs. If there is any other workaround in Jupyter that could be used to circumvent the problem of repeated XML IDs, let me know.

**Comment by agoose77** [comment]
Here's a reproduce:
https://mybinder.org/v2/gist/agoose77/9bce8bcbc3ca393d216c9afce3efdcc4/HEAD?urlpath=lab/tree/reproducer.ipynb

**Comment by agoose77** [comment]
Upon further inspection, this is _not_ a bug with `python-igraph`: I can reproduce this with any HTML using the isolated keyword.
[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gist/agoose77/e39f7361554622575e68da3677caa904/HEAD?urlpath=lab%2Ftree%2Freproducer.ipynb)

**Comment by ntamas** [comment]
Additional info: I could not reproduce on macOS, neither with Chrome 90.0.4430.72, nor with Firefox.

**Comment by agoose77** [comment]
@ntamas can you reproduce the issue on Chrome,macOS with this binder?
[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gist/agoose77/e39f7361554622575e68da3677caa904/HEAD?urlpath=lab%2Ftree%2Freproducer.ipynb)

**Comment by ntamas** [comment]
Yes, I can -- but strangely enough, I need to wait a few seconds before the bug is triggered. I'm trying to make a screen recording to illustrate it.

**Comment by krassowski** [comment]
I started experiencing something similar a few days ago. It's extremely perplexing and annoying.

- I have a notebook of ~80 cells; no iframes at all.
- After loading that notebook the rendering of UI inside of notebook starts to spawn recursively 
- In a runaway process it crashes my Chrome window where JupyterLab is open with SIGTRAP error.
- When I try to narrow down the issue by splitting notebook in half, it is gone.
- This behaviour was not there before. I worked with exact same notebook about a week ago seamlessly.

Cannot share the notebook as-is, but will try to get a reproducible example soon.

**Comment by krassowski** [comment]
Narrowed it down it was an SVG with

```json
"metadata": {
  "image/svg+xml": {
  "isolated": true
}
```

indeed. Notebook fork bomb attached.

[Explode.zip](https://github.com/jupyterlab/jupyterlab/files/6347187/Explode.zip)

**Comment by bdeck8317** [comment]
Also having this issue :( 

Same as others with igraph but from the looks of it it isn't the module but rather a chrome and windows issue?

Any resolutions?

**Comment by ntamas** [comment]
As a workaround, you can try a different browser (it was reported that it works in Firefox), or patch the `_repr_svg_()` method of `Plot` objects in igraph not to use an isolated renderer; see more details [here](https://github.com/igraph/python-igraph/issues/388#issuecomment-823069148). The downside is that by not using an isolated renderer, vertex labels will be rendered incorrectly if you have more than one plot in the workbook.

**Comment by bdeck8317** [comment]
@ntamas cool! That worked.

**Comment by krassowski** [comment]
Also happens in Edge Version 92.0.873.1.

**Comment by krassowski** [comment]
The following fragment in JupyterLab codebase seems to trigger this behaviour:

https://github.com/jupyterlab/jupyterlab/blob/b63af09b5445e8762825d9f0fd86f70dac30e62f/packages/outputarea/src/widget.ts#L1046-L1051

CC @blink1073 as author of the code above.

What appears to have happened is that the `window.location` in the iframe must have changed in a recent Chromium version. The example value of `window.location` for JupyterLab is:

```json
{
    "ancestorOrigins": {
        "0": "https://hub-binder.mybinder.ovh"
    },
    "href": "https://hub-binder.mybinder.ovh/user/agoose77-e39f73-e68da3677caa904-6zfejy6j/lab/tree/reproducer.ipynb",
    "origin": "https://hub-binder.mybinder.ovh",
    "protocol": "https:",
    "host": "hub-binder.mybinder.ovh",
    "hostname": "hub-binder.mybinder.ovh",
    "port": "",
    "pathname": "/user/agoose77-e39f73-e68da3677caa904-6zfejy6j/lab/tree/reproducer.ipynb",
    "search": "",
    "hash": ""
}
```

Let's consider a standalone reproduction (this can be further simplified but I kept it this way so it resembles the JupyterLab codebase for easier navigation):

```js
iframe = document.querySelector('#test')
iframe.contentDocument.open();
wrapped_node = (
  new DOMParser()
  .parseFromString('<div><a target="_self">hi</a></div>', "text/xml")
  .firstChild
)
iframe.contentDocument.write(wrapped_node.innerHTML);
iframe.contentDocument.close();

this_node = iframe
win = this_node.contentWindow;

console.log(win.location)
if (win) {
   win.location.reload();
}
```

1. It causes a 404 on JS Fiddle: https://jsfiddle.net/08mz9uos/4/ - please note that "hi" is still there (at the very end)
2. It does not cause any issue on JS Bin: https://jsbin.com/vikotexoyi/5/edit?html,css,js,output

Comparing the win.location we can see that JS bin changes the location to prevent this issue (`https://null.jsbin.com`):

```json
{
    "ancestorOrigins": {
        "0": "https://null.jsbin.com",
        "1": "https://null.jsbin.com",
        "2": "https://jsbin.com"
    },
    "href": "https://null.jsbin.com/runner",
    "origin": "https://null.jsbin.com",
    "protocol": "https:",
    "host": "null.jsbin.com",
    "hostname": "null.jsbin.com",
    "port": "",
    "pathname": "/runner",
    "search": "",
    "hash": ""
}
```

Which is not really the case for JS Fiddle; interestingly its `win.location` is empty:

```json
{}
```

but apparently chromium auto-populates it with some default values:

![Screenshot from 2021-05-03 18-08-17](https://user-images.githubusercontent.com/5832902/116908259-96062d80-ac3a-11eb-8720-86f49bf8eda1.png)


For comparison, this is JS Bin:

![Screenshot from 2021-05-03 18-08-47](https://user-images.githubusercontent.com/5832902/116908311-a4ece000-ac3a-11eb-9341-024ee93802a0.png)

**Comment by krassowski** [comment]
This comment: https://bugs.chromium.org/p/chromium/issues/detail?id=1102909#c4 looks similar, but it seems that it was like this since Chrome 83 so maybe something else has changed since.

**Comment by krassowski** [comment]
I was able to trigger the issue on Firefox Nightly (90.0a1) via debugger by doing **nothing**. The only action taken was waiting on a breakpoint which introduced a delay; it looks like there might be a race condition which only kicks in Chromium browsers but not in Firefox (unless we set a breakpoint).

![repro-on-ff](https://user-images.githubusercontent.com/5832902/116921535-06697a80-ac4c-11eb-8d26-9a61a2394e70.gif)

**Comment by krassowski** [comment]
If anyone needs to open a notebook with an SVG file in the meantime, this modification works for me:

```bash
sed -i 's/"isolated": true/"isolated": false/' Notebook.ipynb
```
